src/chrome/chromeDebugSession.ts

```diff
@@ -47,9 +48,14 @@ function isChromeError(e: RequestHandleError): e is IChromeError {
     return !!(<IChromeError>e).data;
 }
 
-export class ChromeDebugSession extends LoggingDebugSession {
-    private _debugAdapter: IDebugAdapter;
+export class ChromeDebugSession extends LoggingDebugSession implements ObservableEvents<StepStartedEventsEmitter & NavigatedToUserRequestedUrlEventsEmitter> {
+    private _debugAdapter: IDebugAdapter & ObservableEvents<StepStartedEventsEmitter & NavigatedToUserRequestedUrlEventsEmitter>;
     private _extensionName: string;
+    public readonly Events: StepProgressEventsEmitter;
+    private reporter = new ExecutionTimingsReporter();
+    private haveLaunchExecutionTimingsBeenReported = false;
+
+    public static readonly NavigatedToUserRequestedUrlEventName = 'navigatedToUserRequestedUrl';
```

Comment:
Where is this fired from? And it's not clear that we need interfaces for every event that will be fired.

Reply:
This will get fired from onFrameNavigated in vscode-chrome-debug.

Reply:
I don't think it should be referenced from here then. Navigating is a chrome-debug-specific concept, not something that -core needs to know about.